language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -10,498 +10,238 @@ import System.Environment.Blank
-      removeDirectoryRecursive charliesRestaurantTs
-      assertBool "'charlies-restaurant-ts' should not exist." . not =<< doesDirectoryExist charliesRestaurantTs
-      createDirectoryIfMissing True charliesRestaurantTs
-      assertBool "'charlies-restaurant-ts' should exist." =<< doesDirectoryExist charliesRestaurantTs
-      daml2tsProject [charliesRestaurantDar, groverDar] charliesRestaurantTsSrc
-      assertBool "'CharliesRestaurant.ts' was not created." =<< doesFileExist (charliesRestaurantTsSrc </> "charlies-restaurant-1.0" </> "CharliesRestaurant.ts")
-      assertBool "'packageId.ts' was not created." =<< doesFileExist (charliesRestaurantTsSrc </> "charlies-restaurant-1.0" </> "packageId.ts")
-      assertBool "'Grover.ts' was not created." =<< doesFileExist (charliesRestaurantTsSrc </> "grover-1.0" </> "Grover.ts")
-      assertBool "'packageId.ts' was not created." =<< doesFileExist (charliesRestaurantTsSrc </> "grover-1.0" </> "packageId.ts")
-      step "Compiling 'charlies-restaurant-ts' to JavaScript... "
-      writePackageConfigs charliesRestaurantTs
-      withCurrentDirectory charliesRestaurantTs $ do
-        yarnProject ["run", "build"]
-        assertBool "'Grover.js' was not created." =<< doesFileExist (charliesRestaurantTsLib </> "grover-1.0" </> "Grover.js")
-        assertBool "'CharliesRestaurant.js' was not created." =<< doesFileExist (charliesRestaurantTsLib </> "charlies-restaurant-1.0" </> "CharliesRestaurant.js")
-        step "Linting 'charlies-restaurant' ... "
-        yarnProject ["run", "lint"]
-
-  , testCaseSteps "Package name collision test" $ \step -> withTempDir $ \tmpDir -> do
-      let grover = tmpDir </> "grover"
-          groverDaml = grover </> "daml"
-          groverTs = tmpDir </> "grover-ts"
-          groverDar = grover </> ".daml" </> "dist" </> "grover-1.0.dar"
-      step "Creating project 'grover'..."
-      createDirectoryIfMissing True groverDaml
-      createDirectoryIfMissing True groverTs
-      writeFileUTF8 (grover </> "daml" </> "Grover.daml") $ unlines
-        [ "daml 1.2"
-        , "module Grover where"
-        , "template Grover"
-        , "  with puppeteer : Party"
-        , "  where"
-        , "    signatory puppeteer"
-        , "    choice Grover_GoSuper: ContractId Grover"
-        , "      controller puppeteer"
-        , "      do"
-        , "        return self"
-        ]
-      writeFileUTF8 (grover </> "daml.yaml") $ unlines
-        [ "sdk-version: " <> sdkVersion
-        , "name: grover"
-        , "version: \"1.0\""
-        , "source: daml"
-        , "exposed-modules: [Grover]"
-        , "dependencies:"
-        , "  - daml-prim"
-        , "  - daml-stdlib"
-        ]
-      buildProject grover []
-      assertBool "grover-1.0.dar was not created." =<< doesFileExist groverDar
-      let elmo = tmpDir </> "elmo"
+      withCurrentDirectory grover $ do
```

Comment:
This is impossible to review, can we please separate refactorings (e.g. factoring out the creation of `daml.yaml` which really should also go into a library since we want it from other tests) from the actual changes that here?